pkg/storage/stores/indexshipper/uploads/table_manager.go

```diff
@@ -130,9 +133,11 @@ func (tm *tableManager) uploadTables(ctx context.Context) {
 
 	level.Info(util_log.Logger).Log("msg", "uploading tables")
 
+	status := statusSuccess
 	for _, table := range tm.tables {
 		err := table.Upload(ctx)
 		if err != nil {
+			status = statusFailure
```

Comment:
Q: can this be confusing when only few tables failed to upload. But in the metrics, we record as though everything is failed?
`tm.metrics.tablesUploadOperationTotal.WithLabelValues(status).Inc()`

Reply:
We track the overall upload operation. We can count individual table upload operations as well but the investigator would anyways have to check the logs to find more details about it.